Update Script Editor to be Unit, Lesson Group and Lesson

dmcavoy opened the pull request:
Moves to having the top level card be Unit, then Lesson Groups, and Lessons are now the draggable token on the script edit page. 

Note: The Script edit gui is still broken here (aka it can't save). This just gets us to the right UI. 

<img width="506" alt="Screen Shot 2020-09-29 at 12 08 20 AM" src="https://user-images.githubusercontent.com/208083/94859939-59075d80-0403-11eb-8243-94b8a17c39d9.png">

## Links

<!--
  Any relevant links to external resources; ie, specification documents, jira
  items, related PRs, honeybadger errors, etc
-->

- [PLAT-268](https://codedotorg.atlassian.net/browse/PLAT-268)
- [PLAT-262](https://codedotorg.atlassian.net/browse/PLAT-262)

## Testing story

Updated tests to match the new updated components.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

davidsbailey commented:
@dmcavoy it looks like there are quite a few files being added and removed. Are any of the new files moved/renamed copies of the deleted files? 

Also, from the screenshot, it seems like we might want to take "Levels" out of the "Lessons and Levels" heading, but I'm fine with keeping it if that's intentional.

dmcavoy commented:
> @dmcavoy it looks like there are quite a few files being added and removed. Are any of the new files moved/renamed copies of the deleted files?

@davidsbailey - Yes you are right there are moved and renamed files. I actually redid this whole PR to explicitly `git mv` everything and it didn't seem to work. Let me giving you a mapping to hopefully help

| Object | Old File | New File|
| -- | -- | --|
| Container for all the Blue Cards | LessonGroups.jsx | None - no longer needed since only one blue card|
| Blue Cards | LessonGroupCard.jsx | UnitCard.jsx|
| White Card| LessonCard.jsx | LessonGroupCard.jsx|
| Token | None | LessonToken.jsx|


NewLessonGroupInput.jsx - Is no longer needed. It was the way we were adding new lesson groups but I found a better way to do it.

> Also, from the screenshot, it seems like we might want to take "Levels" out of the "Lessons and Levels" heading, but I'm fine with keeping it if that's intentional.

Thats a good catch. I can update that.

dmcavoy commented:
<img width="1004" alt="Screen Shot 2020-10-02 at 1 01 20 PM" src="https://user-images.githubusercontent.com/208083/94949712-611cd700-04af-11eb-9e48-17939170d137.png">

davidsbailey commented:
Thanks Dani! 👀   looking...

davidsbailey commented on apps/test/unit/lib/levelbuilder/script-editor/UnitCardTest.jsx:
```diff
@@ -0,0 +1,58 @@
+
+describe('UnitCard', () => {
+  it('displays UnitCard correctly', () => {
+    let wrapper = shallow(<UnitCard {...defaultProps} />);
```
before January launch, this component should have a test which uses `mount` to render LessonGroup and Lesson cards within it.

dmcavoy replied:
Tracked: https://codedotorg.atlassian.net/browse/PLAT-375

davidsbailey reviewed (approved):
Looks good! I see two areas for future cleanup here:
* PLAT-23 fix dragging bugs by moving from react-motion to react-draggable-list
* PLAT-374 move Script Editor off of position-based logic
